lib/javatests/com/google/android/material/imageview/ShapeableImageViewTest.java

```diff
@@ -0,0 +1,292 @@
+    assertThat(imageView.getPaddingBottom()).isEqualTo(0);
+
+    forceResolveLayoutDirection(imageView);
+    imageView.onMeasure(WIDTH_MEASURE_SPEC, HEIGHT_MEASURE_SPEC);
+
+    assertThat(imageView.getContentPaddingStart()).isEqualTo(1);
+    assertThat(imageView.getContentPaddingTop()).isEqualTo(2);
+    assertThat(imageView.getContentPaddingEnd()).isEqualTo(3);
+    assertThat(imageView.getContentPaddingBottom()).isEqualTo(4);
+    assertThat(imageView.getPaddingStart()).isEqualTo(0);
+    assertThat(imageView.getPaddingTop()).isEqualTo(0);
+    assertThat(imageView.getPaddingEnd()).isEqualTo(0);
+    assertThat(imageView.getPaddingBottom()).isEqualTo(0);
+  }
+
+  @Config(sdk = {16, 18, 19, 21, 30})
+  @Test
+  public void setPaddingRelativeBeforeMeasure() {
+    ShapeableImageView imageView = new ShapeableImageView(context);
+
+    imageView.setPaddingRelative(1, 2, 3, 4);
+
+    assertThat(imageView.getContentPaddingStart()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingTop()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingEnd()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingBottom()).isEqualTo(0);
+    assertThat(imageView.getPaddingStart()).isEqualTo(1);
+    assertThat(imageView.getPaddingTop()).isEqualTo(2);
+    assertThat(imageView.getPaddingEnd()).isEqualTo(3);
+    assertThat(imageView.getPaddingBottom()).isEqualTo(4);
+
+    forceResolveLayoutDirection(imageView);
+    imageView.onMeasure(WIDTH_MEASURE_SPEC, HEIGHT_MEASURE_SPEC);
+
+    assertThat(imageView.getContentPaddingStart()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingTop()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingEnd()).isEqualTo(0);
+    assertThat(imageView.getContentPaddingBottom()).isEqualTo(0);
+    assertThat(imageView.getPaddingStart()).isEqualTo(1);
+    assertThat(imageView.getPaddingTop()).isEqualTo(2);
+    assertThat(imageView.getPaddingEnd()).isEqualTo(3);
+    assertThat(imageView.getPaddingBottom()).isEqualTo(4);
+  }
+
+  @Config(sdk = {16, 18, 19, 21, 30})
+  @Test
+  public void setPaddingAndContentPaddingRelativeBeforeMeasure() {
+    ShapeableImageView imageView = new ShapeableImageView(context);
+
+    imageView.setPaddingRelative(1, 2, 3, 4);
+    imageView.setContentPaddingRelative(5, 6, 7, 8);
+
+    assertThat(imageView.getContentPaddingStart()).isEqualTo(5);
+    assertThat(imageView.getContentPaddingTop()).isEqualTo(6);
+    assertThat(imageView.getContentPaddingEnd()).isEqualTo(7);
+    assertThat(imageView.getContentPaddingBottom()).isEqualTo(8);
+    assertThat(imageView.getPaddingStart()).isEqualTo(1);
+    assertThat(imageView.getPaddingTop()).isEqualTo(2);
+    assertThat(imageView.getPaddingEnd()).isEqualTo(3);
+    assertThat(imageView.getPaddingBottom()).isEqualTo(4);
+
+    forceResolveLayoutDirection(imageView);
+    imageView.onMeasure(WIDTH_MEASURE_SPEC, HEIGHT_MEASURE_SPEC);
+
+    assertThat(imageView.getContentPaddingStart()).isEqualTo(5);
```

Comment:
Can we also add tests to check if super.getPaddingTop/Bottom/Start/End() returns correct values?